processor/src/main/java/org/mapstruct/ap/internal/model/source/TargetReference.java

```diff
@@ -246,9 +247,22 @@ else if ( targetWriteAccessor == null ) {
             }
         }
 
-        private void reportMappingError(Message msg, Object... objects) {
-            messager.printMessage( method.getExecutable(), mapping.getMirror(), mapping.getSourceAnnotationValue(),
-                msg, objects );
+        private void reportMappingError() {
+            if ( errorMessage == Message.BEANMAPPING_UNKNOWN_PROPERTY_IN_RESULTTYPE ) {
```

Comment:
This is a bit connected to my answer in https://github.com/mapstruct/mapstruct/pull/1168#issuecomment-300857751